feature_engine/encoding/count_frequency.py

```diff
@@ -143,22 +175,87 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         y: pandas Series, default = None
             y is not needed in this encoder. You can pass y or None.
+
+        threshold : int/float if encoding is "count" and "frequency", default = None
+            Thresholds for each column needed for binning infrequent categories
         """
+
         X = check_X(X)
         self._check_or_select_variables(X)
         self._get_feature_names_in(X)
 
         self.encoder_dict_ = {}
 
-        # learn encoding maps
-        for var in self.variables_:
+        if self.threshold is not None and self.encoding_method == "count":
+            # learn encoding maps
+            for var in self.variables_:
+                temp = {"backoff_bin": 0}
+                for k, v in X[var].value_counts().to_dict().items():
+                    if v <= self.threshold:
+                        temp["backoff_bin"] = temp["backoff_bin"] + v
+                    else:
+                        temp[k] = v
+                self.encoder_dict_[var] = temp
+
+        elif self.threshold is not None and self.encoding_method == "frequency":
+            for var in self.variables_:
+                temp = {"backoff_bin": 0}
+                n_obs = float(len(X))
+                for k, v in (X[var].value_counts() / n_obs).to_dict().items():
+                    if v <= self.threshold:
+                        temp["backoff_bin"] = temp["backoff_bin"] + v
+                    else:
+                        temp[k] = v
+                self.encoder_dict_[var] = temp
+
+        else:
+
             if self.encoding_method == "count":
-                self.encoder_dict_[var] = X[var].value_counts().to_dict()
+                for var in self.variables_:
+                    self.encoder_dict_[var] = X[var].value_counts().to_dict()
 
             elif self.encoding_method == "frequency":
-                n_obs = float(len(X))
-                self.encoder_dict_[var] = (X[var].value_counts() / n_obs).to_dict()
+                for var in self.variables_:
+                    n_obs = float(len(X))
+                    self.encoder_dict_[var] = (X[var].value_counts() / n_obs).to_dict()
 
         self._check_encoding_dictionary()
 
         return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Replace the categories with counts/frequencies.Replace infrequent categories by
```

Comment:
I would not modify this docstring. Can we revert to the original? all main info goes in the init docstrings at the top of the class.

More conventions :/

If you think it is important, then move to the top. we can also add an example in the user_guide: https://github.com/feature-engine/feature_engine/blob/main/docs/user_guide/encoding/CountFrequencyEncoder.rst

Reply:
> 
Yes, adding about threshold again in transform would be repetitive since its already present in init, I'll revert back to original.